s2fft/precompute_transforms/construct.py

```diff
@@ -53,27 +60,102 @@ def spin_spherical_kernel(
             "Reality acceleration only supports spin 0 fields. "
             + "Defering to complex transform."
         )
+    if mode.lower() == "price_mcewen" and abs(spin) >= 5:
+        mode = "risbo"
+        warn(
+            "The Price-McEwen can become unstable for spins >= 5. "
+            + "Defaulting to slower but more stable Risbo recursions."
+        )
+
+    if mode.lower() == "auto":
+        # This mode automatically determines which recursion is best suited for the
+        # current parameter configuration.
+        mode = "risbo" if abs(spin) >= 5 else "price_mcewen"
+
+    dl = []
     m_start_ind = L - 1 if reality else 0
     m_dim = L if reality else 2 * L - 1
 
+    # Determine theta locations for forward vs inverse transform.
     if forward and sampling.lower() in ["mw", "mwss"]:
         sampling = "mwss"
         thetas = samples.thetas(2 * L, "mwss")
     else:
         thetas = samples.thetas(L, sampling, nside)
 
-    dl = np.zeros((len(thetas), L, m_dim), dtype=np.float64)
-    for t, theta in enumerate(thetas):
-        for el in range(abs(spin), L):
-            dl[t, el] = recursions.turok.compute_slice(theta, el, L, -spin, reality)[
-                m_start_ind:
-            ]
-            dl[t, el] *= np.sqrt((2 * el + 1) / (4 * np.pi))
+    # Calculate Wigner d-function elements through the Price-McEwen recursion.
+    # - The complexity of this approach is O(L^3).
+    # - This approach becomes inaccurate for abs(spins) >= 5.
+    if mode.lower() == "price_mcewen":
+        dl = np.zeros((len(thetas), L, m_dim), dtype=np.float64)
+        for t, theta in enumerate(thetas):
+            for el in range(abs(spin), L):
+                dl[t, el] = recursions.turok.compute_slice(
+                    theta, el, L, -spin, reality
+                )[m_start_ind:]
+                dl[t, el] *= np.sqrt((2 * el + 1) / (4 * np.pi))
+
+    # Calculate Wigner d-function elements through the Risbo recursion.
+    elif mode.lower() == "risbo":
```

Comment:
After this `elif` we should have a final `else` that catches if an unsupported recursions is provided.